forest/cli/src/cli/send_cmd.rs

```diff
@@ -7,9 +7,81 @@ use forest_json::message::json::MessageJson;
 use forest_rpc_client::{mpool_push_message, wallet_default_address};
 use fvm_shared::{address::Address, econ::TokenAmount, message::Message, METHOD_SEND};
 use num::BigInt;
+use rust_decimal::prelude::*;
+use rust_decimal_macros::dec;
 
 use super::{handle_rpc_err, Config};
 
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct FILAmount {
+    pub value: TokenAmount,
+}
+
+/// `FILAmount::from_str` adds support for parsing a string (input to the
+/// 'amount' field of the the `send` command) with all units supported by forest
+/// wallet ("attoFIL", "femtoFIL", "picoFIL", "nanoFIL", "microFIL", "milliFIL",
+/// and "FIL") and converting the amount into an attoFIL `TokenAmount`. To match
+/// the current behavior in Lotus, the default units (if no units are specified
+/// in the command line) are "FIL".
+impl FromStr for FILAmount {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        fn error_call<T: std::fmt::Display>(s: &str, e: T) -> anyhow::Error {
+            anyhow::anyhow!("failed to parse fil amount: {}. {}.", &s, e)
+        }
+
+        let suffix_idx = s
+            .rfind(char::is_numeric)
+            .ok_or_else(|| anyhow::anyhow!("failed to parse fil amount: {}. No digits.", s))?;
+        let (val, suffix) = s.split_at(suffix_idx + 1);
+
+        // string length check to match Lotus logic
+        if val.chars().count() > 50 {
+            return Err(error_call(
+                s,
+                anyhow::Error::msg(format!("string length too large: {}", val.chars().count())),
+            ));
+        }
+
+        let suffix = suffix.trim().to_lowercase();
+        let multiplier = match suffix.strip_suffix("fil").map(str::trim).unwrap_or(&suffix) {
+            "atto" | "a" => dec!(1e0),
+            "femto" => dec!(1e3),
+            "pico" => dec!(1e6),
+            "nano" => dec!(1e9),
+            "micro" => dec!(1e12),
+            "milli" => dec!(1e15),
+            "" => dec!(1e18),
+            _ => {
+                return Err(error_call(
+                    s,
+                    anyhow::Error::msg(format!("unrecognized suffix: {}", suffix)),
+                ));
+            }
+        };
+
+        let parsed_val = Decimal::from_str(val).map_err(|e| error_call(s, e))?;
+
+        let val = parsed_val * multiplier;
+        if val.normalize().scale() > 0 {
+            return Err(error_call(
+                s,
+                anyhow::Error::msg(format!("{} must convert to a whole attoFIL value", &val)),
+            ));
+        }
+        let attofil_val = val.trunc().to_u128().ok_or(error_call(s, anyhow::Error::msg(
+            "Could not covert amount input to send into an attoFIL value (note that negative FIL amounts are not valid)"),
+        ))?;
+
+        let token_amount = TokenAmount::from_atto(BigInt::from(attofil_val));
```

Comment:
You don't need the `BigInt` and `token_amount` is only used once so you can just inline it.